File: src/main/java/model/EmailVerificationToken.java
package model;

import jakarta.persistence.*;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.Nationalized;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "EmailVerificationTokens")
public class EmailVerificationToken {
    @Id
    @ColumnDefault("newid()")
    @Column(name = "TokenID", nullable = false)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "UserID", nullable = false)
    private User userID;

    @ColumnDefault("newid()")
    @Column(name = "Token")
    private UUID token;

    @Nationalized
    @Column(name = "TokenType", nullable = false, length = 50)
    private String tokenType;

    @Column(name = "ExpiryDate", nullable = false)
    private Instant expiryDate;

    @ColumnDefault("0")
    @Column(name = "IsUsed")
    private Boolean isUsed;

    @ColumnDefault("getdate()")
    @Column(name = "CreatedDate")
    private Instant createdDate;

    @Column(name = "UsedDate")
    private Instant usedDate;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public User getUserID() {
        return userID;
    }

    public void setUserID(User userID) {
        this.userID = userID;
    }

    public UUID getToken() {
        return token;
    }

    public void setToken(UUID token) {
        this.token = token;
    }

    public String getTokenType() {
        return tokenType;
    }

    public void setTokenType(String tokenType) {
        this.tokenType = tokenType;
    }

    public Instant getExpiryDate() {
        return expiryDate;
    }

    public void setExpiryDate(Instant expiryDate) {
        this.expiryDate = expiryDate;
    }

    public Boolean getIsUsed() {
        return isUsed;
    }

    public void setIsUsed(Boolean isUsed) {
        this.isUsed = isUsed;
    }

    public Instant getCreatedDate() {
        return createdDate;
    }

    public void setCreatedDate(Instant createdDate) {
        this.createdDate = createdDate;
    }

    public Instant getUsedDate() {
        return usedDate;
    }

    public void setUsedDate(Instant usedDate) {
        this.usedDate = usedDate;
    }

}

File: src/main/java/model/EmailVerificationToken.kt
package model

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault
import org.hibernate.annotations.Nationalized
import java.time.Instant
import java.util.*

@Entity
@Table(name = "EmailVerificationTokens")
open class EmailVerificationToken {
    @Id
    @ColumnDefault("newid()")
    @Column(name = "TokenID", nullable = false)
    open var id: UUID? = null

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "UserID", nullable = false)
    open var userID: User? = null

    @ColumnDefault("newid()")
    @Column(name = "Token")
    open var token: UUID? = null

    @Nationalized
    @Column(name = "TokenType", nullable = false, length = 50)
    open var tokenType: String? = null

    @Column(name = "ExpiryDate", nullable = false)
    open var expiryDate: Instant? = null

    @ColumnDefault("0")
    @Column(name = "IsUsed")
    open var isUsed: Boolean? = null

    @ColumnDefault("getdate()")
    @Column(name = "CreatedDate")
    open var createdDate: Instant? = null

    @Column(name = "UsedDate")
    open var usedDate: Instant? = null
}